x-pack/plugins/maps/public/components/map/mb/view.js

```diff
@@ -357,33 +358,48 @@ export class MBMapContainer extends React.Component {
     }
   }
 
-  _renderContentToTooltip(content, location) {
+  _showTooltip() {
     if (!this._isMounted) {
       return;
     }
     const isLocked = this.props.tooltipState.type === TOOLTIP_TYPE.LOCKED;
     ReactDOM.render((
       <FeatureTooltip
-        properties={content}
+        tooltipState={this.props.tooltipState}
+        loadFeatureProperties={this._loadFeatureProperties}
         closeTooltip={this._onTooltipClose}
         showFilterButtons={this.props.isFilterable && isLocked}
         showCloseButton={isLocked}
       />
     ), this._tooltipContainer);
 
-    this._mbPopup.setLngLat(location)
+    this._mbPopup.setLngLat(this.props.tooltipState.location)
       .setDOMContent(this._tooltipContainer)
       .addTo(this._mbMap);
   }
 
-
-  async _showTooltip()  {
+  _loadFeatureProperties = async ({ layerId, featureId }) => {
     const tooltipLayer = this.props.layerList.find(layer => {
-      return layer.getId() === this.props.tooltipState.layerId;
+      return layer.getId() === layerId;
     });
+    if (!tooltipLayer) {
+      throw new Error(
+        i18n.translate('xpack.maps.mb.unableToFindLayerMsg', {
+          defaultMessage: 'Unable to find layer {layerId}',
+          values: { layerId }
+        })
+      );
+    }
     const targetFeature = tooltipLayer.getFeatureById(this.props.tooltipState.featureId);
-    const formattedProperties = await tooltipLayer.getPropertiesForTooltip(targetFeature.properties);
-    this._renderContentToTooltip(formattedProperties, this.props.tooltipState.location);
+    if (!targetFeature) {
+      throw new Error(
```

Comment:
I wouldn't throw, rather fail silently. This error code-path should never occur (for every feature we have put on the map, getFeatureById should return an object).  If it does throw, it is a bug in our code.

Maybe it can occur in the brief moment between data being updated in our store, and it not having been synced yet on the mapbox-map (since we do a slight debounce). But that would be expected, and then we should just return silently.